src/main/java/seedu/address/model/export/ExportPathFactory.java

```diff
@@ -0,0 +1,32 @@
+//@@author LeowWB
+
+package seedu.address.model.export;
+
+/**
+ * Class used for creating of {@code ExportPath}s. Follows the factory pattern.
+ */
+public class ExportPathFactory {
+
+    public static final String MESSAGE_ILLEGAL_PATH = "The file path you have provided is not allowed.\n"
+            + "File paths must consist entirely of alphanumeric characters, spaces, and the following:\n"
+            + "~\\/-_!:[]()\n"
+            + "It must also have one of the following extensions:\n"
+            + "'.docx', '.json'";
+
+    /**
+     * Creates an {@code ExportPath} from a given String.
+     *
+     * @param exportPath String representing the path of the file that we plan to export to
+     * @return ExportPath that represents the same path as the given String
+     * @throws IllegalArgumentException if the given String does not match the format of any of the ExportPaths
+     */
+    public static ExportPath getExportPath(String exportPath) throws IllegalArgumentException {
+        if (DocumentPath.isValid(exportPath)) {
+            return new DocumentPath(exportPath);
+        } else if (JsonExportFilePath.isValid(exportPath)) {
+            return new JsonExportPath(exportPath);
+        } else {
+            throw new IllegalArgumentException(MESSAGE_ILLEGAL_PATH);
+        }
```

Comment:
Perhaps using guard clauses here would be better

Reply:
I don't think so? Guard clauses are typically used to handle exceptional cases that the code is not particularly interested in. The lines you mentioned, though, _are_ the main functionality of that method.